### DISCUSSION TITLE
antd从4升级到5，umi从3升级到 4，build失败

### DISCUSSION THREAD

**Discussion by Emensionyu** [umijs/umi#12237] [Q&A]
Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
info  - Memory Usage: 909.54 MB (RSS: 1158.28 MB)
fatal - Error: ERROR in ./src/components/Modal/index.tsx 85:0-38
export 'ModalProps' (reexported as 'ModalProps') was not found in 'antd/lib/modal' (possible exports: __esModule, default)
 @ ./src/components/index.ts 6:0-28 20:0-203
 @ ./src/layouts/index.tsx
 @ ./src/.umi-production/core/route.tsx
 @ ./src/.umi-production/umi.ts 10:0-41 27:17-26

ERROR in ./src/components/Modal/index.tsx 85:0-38
export 'ModalFuncProps' (reexported as 'ModalFuncProps') was not found in 'antd/lib/modal' (possible exports: __esModule, default)
 @ ./src/components/index.ts 6:0-28 20:0-203
 @ ./src/layouts/index.tsx
 @ ./src/.umi-production/core/route.tsx
 @ ./src/.umi-production/umi.ts 10:0-41 27:17-26

webpack compiled with 2 errors

**Comment by fz6m** [comment]
如果你要把 umi 3 的项目升级为 umi 4 ，就不要动 antd 和其他非 umi 相关库的版本，因为 antd 4 到 5 本来就是有区别的，所以两个升级混在一起就不知道是哪里导致的问题了。

在 umi 3 升级到 4 的时候，最好新初始化一个 umi 4 的模板项目，参考已有的各种文档，在 issue 或 discussion 搜索关键词看看有没有帮助的信息，同时搜过搜索引擎搜索关键词解决一些问题，升级必然要修改一些东西，通过一点一点往里搬的方式升级比较容易。
